### ISSUE TITLE
Do not prettify atomic vectors

### ISSUE THREAD

**Issue by yihui** [jeroenooms/jsonlite#84]
This may be a purely aesthetic issue, but you know... why there is no love for XML... :smiling_imp: 

Personally I do not like vectors to be prettified by adding line breaks after _every_ element. This makes the data less readable to human eyes (well, my eyes). I love the RJSONIO behavior, which is to add the line break only after a row or column vector:

```r
> jsonlite::toJSON(matrix(1:12, 4))
[[1,5,9],[2,6,10],[3,7,11],[4,8,12]] 
> jsonlite::toJSON(matrix(1:12, 4), pretty=TRUE)
[
    [
        1,
        5,
        9
    ],
    [
        2,
        6,
        10
    ],
    [
        3,
        7,
        11
    ],
    [
        4,
        8,
        12
    ]
]
> cat(RJSONIO::toJSON(matrix(1:12, 4)))
[ [ 1, 5, 9 ],
[ 2, 6, 10 ],
[ 3, 7, 11 ],
[ 4, 8, 12 ] ]
```

Currently you either get a terribly wide single line of data and you can never make sense of the structure, or a terribly long page of data and you get lost as you scroll down the page, when the data is moderately large. So I'm filing this request to make `pretty = TRUE` prettier :)

**Comment by jeroenooms** [comment]
I don't think yajl exposes a simple way to control this. It's also a bit ambiguous; how would you expect an heterogeneous array to be indented?

```json
[ 1, 2, [ 3, 4 ], { "foo" : "bar" } ]
```

**Comment by yihui** [comment]
For heterogeneous arrays (lists), I think it is fine that each element starts a new line.

```json
[
  1,
  2,
  [ 3, 4 ],
  {
    "foo" : "bar" 
  }
]
```

In R, we often work with homogeneous objects such as matrices and data frames. I hope these objects  can be printed in a pretty fashion.

**Comment by jeroenooms** [comment]
That would require that the pretty printing is implemented in R so that it can distinguish between atomic vectors and lists. Currently the pretty printing is done in yajl, entirely separate form the toJSON conversion. The prettify function takes json as input, so it cannot have different behavior for different R types.

I'll leave the issue open for know but with the current implementation this would be very difficult I am afraid :(

**Comment by yihui** [comment]
During the final round of collapsing, can you insert `\n` among rows/columns? https://github.com/jeroenooms/jsonlite/blob/7bb5e6b/R/asJSON.data.frame.R#L95-L99 RJSONIO has the `collapse` argument that allows you to specify the separator among lines, and I think it is very useful. I can live without indentation, and it is much harder to tolerate putting everything in one giant single line, e.g. it is hard to see what exactly changed in version control.

**Comment by yihui** [comment]
Closing this issue since it has been implemented in jsonlite 0.9.16.
